operate/services/manage.py

```diff
@@ -323,53 +363,92 @@ def deploy_service_onchain(  # pylint: disable=too-many-statements
         """
-        self.logger.info("Loading service")
         service = self.load_or_create(hash=hash)
-        user_params = service.chain_data.user_params
+        for chain_id in service.chain_configs.keys():
+            self._deploy_service_onchain_from_safe(
+                hash=hash,
+                chain_id=chain_id,
+            )
+
+    def _deploy_service_onchain_from_safe(  # pylint: disable=too-many-statements,too-many-locals
+        self,
+        hash: str,
+        chain_id: str,
+    ) -> None:
+        """
+        Deploy as service on-chain
+
+        :param hash: Service hash
+        """
+
+        self.logger.info(f"_deploy_service_onchain_from_safe {chain_id=}")
+        service = self.load_or_create(hash=hash)
+        chain_config = service.chain_configs[chain_id]
+        ledger_config = chain_config.ledger_config
+        chain_data = chain_config.chain_data
+        user_params = chain_config.chain_data.user_params
         keys = service.keys
         instances = [key.address for key in keys]
-        wallet = self.wallet_manager.load(service.ledger_config.type)
-        sftxb = self.get_eth_safe_tx_builder(service=service)
-        if user_params.use_staking and not sftxb.staking_slots_available(
-            staking_contract=STAKING[service.ledger_config.chain]
-        ):
-            raise ValueError("No staking slots available")
+        wallet = self.wallet_manager.load(ledger_config.type)
+        sftxb = self.get_eth_safe_tx_builder(ledger_config=ledger_config)
 
-        if service.chain_data.token > -1:
+        # TODO fix this
+        os.environ["CUSTOM_CHAIN_RPC"] = ledger_config.rpc
+        os.environ["OPEN_AUTONOMY_SUBGRAPH_URL"] = "https://subgraph.autonolas.tech/subgraphs/name/autonolas-staging"
+
+        current_agent_id = None
+        if chain_data.token > -1:
             self.logger.info("Syncing service state")
-            info = sftxb.info(token_id=service.chain_data.token)
-            service.chain_data.on_chain_state = OnChainState(info["service_state"])
-            service.chain_data.instances = info["instances"]
-            service.chain_data.multisig = info["multisig"]
+            info = sftxb.info(token_id=chain_data.token)
+            chain_data.on_chain_state = OnChainState(info["service_state"])
+            chain_data.instances = info["instances"]
+            chain_data.multisig = info["multisig"]
+            current_agent_id = info["canonical_agents"][0]  # TODO Allow multiple agents
             service.store()
-        self.logger.info(f"Service state: {service.chain_data.on_chain_state.name}")
+        self.logger.info(f"Service state: {chain_data.on_chain_state.name}")
+
+        if user_params.use_staking:
+            staking_params = sftxb.get_staking_params(
+                staking_contract=STAKING[ledger_config.chain][user_params.staking_program_id],
+            )
+        else:  # TODO fix this - using pearl beta params
```

Comment:
Why is this the else condition for `use_staking`? Pearl Beta is also a staking contract.

Reply:
Logically it would be more correct to have `use_staking as true` and then added `if migration is true`?

Reply:
This is a hotfix to avoid having uninitialized variables. If use_staking = false, it will by default initialize with (possible unused) Pearl beta variables.